### PR TITLE
Feature/user profile

### PR DESCRIPTION
**Pull request by tavisca-mrathore**

### CODE REVIEW

File: src/app/services/product.service.ts [hunk 1]
```diff
@@ -55,15 +50,12 @@ export class ProductService implements IproductService {
...
-    if(environment.isPreviewEnabled)
-    {
+  GetPreview(product: Product): GetProductDetailsResponse {
+    if (environment.isPreviewEnabled) {
       return this.productMockService.GetPreview(product);
     }
-    else
-    {
-      
+    else {
```
Comment (Tavisca-smohite) [inline src/app/services/product.service.ts]: remove empty else.
Also actual product service impl should not call mock service's getpreview impl.
if both wants the same implementation then move it to the common class

File: src/app/home-module/advertisment-details/advertisment-details.component.html [hunk 2]
```diff
@@ -22,9 +22,9 @@
         <span>{{productdetails.product.description}}</span>
         <span>Is Negotiable : {{productdetails.product.price.isNegotiable}}</span>
         <span>Pickup Address: {{ productdetails.product.pickupAddress.line1 }},
-                              {{ productdetails.product.pickupAddress.line2 }},
-                              {{ productdetails.product.pickupAddress.city }},
-                              {{ productdetails.product.pickupAddress.state }}
+          {{ productdetails.product.pickupAddress.line2 }},
```
Comment (Tavisca-smohite) [inline src/app/home-module/advertisment-details/advertisment-details.component.html]: `pickupAddress` is optional field in the response , consider adding null checks

Comment (Tavisca-smohite) [reply src/app/home-module/advertisment-details/advertisment-details.component.html]: (elvis operator ?. e.g, `productdetails?.product?.pickupAddress?.line1`

File: src/app/services/product.service.ts [hunk 3]
```diff
@@ -93,18 +85,17 @@ export class ProductService implements IproductService {
       let getProductListUrl: string = this.getUrl("/") + "products/" + productId;
       return this.http.get<GetProductDetailsResponse>(getProductListUrl, {
         headers: this.headers
-      }).pipe(retry(1),catchError(this.errorHandler));
+      }).pipe(retry(1), catchError(this.errorHandler));
```
Comment (Tavisca-smohite) [inline src/app/services/product.service.ts:102]: I would like to understand `getProducts` api as per contracts return you this response :
https://github.com/tavisca-sgogawale/OnlineRetailPortal/blob/develop/Contracts/v1.0/seller/response.json5
according to that:
````json
//Response data
{
	//[Mandatory] id of the product
	"id" : "P123",
	//[Mandatory] title of the product
	"title" : "Watch" ,
	//[Mandatory] description of the product
	"description" : "something something something",
	//[Mandatory] images of the product
	"heroImageUrl" : "www.example.com/imageName",
	//[Mandatory] 
	"price" : 
	{
.
.
.
.
.

````
there is no `product` property in the response nor the seller . 
Please get it validated from people worked on this api . As if seller field is null , then you are displaying the error page.(As per last discussion , we are only returning sellerId in the response.

**Review by Tavisca-smohite — CHANGES_REQUESTED** [review]

File: src/app/services/product.service.ts (hunk 1 above)
Comment (tavisca-mrathore) [reply src/app/services/product.service.ts]: discussed this with nikita.
she left it purposefully to accommodate new changes in the future.
she will be done today then we can proceed to close PR.

File: src/app/home-module/advertisment-details/advertisment-details.component.html (hunk 2 above)
Comment (tavisca-mrathore) [reply src/app/home-module/advertisment-details/advertisment-details.component.html]: had a discussion with aditi.
she will be fixing this today.

File: src/app/services/product.service.ts (hunk 3 above)
Comment (tavisca-mrathore) [reply src/app/services/product.service.ts:102]: had a discussion with nikita.
the code so far is good to go.
the seller contract needs to be updated.

**Review by achirametavisca — APPROVED** [review]

**Pull request merged**